server/py/services/api/initial_data.py

```diff
@@ -132,22 +134,26 @@ def _initialize_db_from_scratch(
 
 
 def _migrate_existing_data(
+    engine: sqlalchemy.engine.Engine,
     perform_migrations_if_needed: bool = False,
 ):
-    # create mysql util, and if mlrun is configured to use mysql, wait for it to be live and set its db modes
-    mysql_util = framework.utils.db.mysql.MySQLUtil(
-        mlrun.utils.logger
-    )  # TODO make this DB agnostic
     alembic_util = _create_alembic_util()
-    if mysql_util.get_mysql_dsn_data():
-        mysql_util.wait_for_db_liveness()
-        mysql_util.set_modes(mlrun.mlconf.httpdb.db.mysql.modes)
+    if engine.dialect in (
+        mlrun.common.db.dialects.Dialects.MYSQL,
+        mlrun.common.db.dialects.Dialects.POSTGRESQL,
+    ):
+        db_util = DBUtil()
+        db_util.wait_for_db_liveness()
+        if engine.name == mlrun.common.db.dialects.Dialects.MYSQL:
+            db_util.set_configurations(
+                config_items=mlrun.mlconf.httpdb.db.mysql_engine.modes,
+            )
         (
             is_migration_needed,
             is_backup_needed,
         ) = _resolve_needed_operations(alembic_util)
     else:
-        dsn = mysql_util.get_dsn()
+        dsn = DBUtil.get_dsn()
         if dsn.startswith(mlrun.common.db.dialects.Dialects.SQLITE):
             mlrun.utils.logger.debug("SQLite DB is used, liveness check not needed")
         else:
```

Comment:
I thought the point of `DBUtil` is to avoid these `engine.dialect` ifs (from line 141 to line 165). Maybe all of this can be abstracted there and there should also be a `SQLiteUtil`? It can have `wait_for_db_liveness`, `set_configurations` and `_resolve_needed_operations` all as NOOPs